editor.planx.uk/src/pages/FlowEditor/lib/analyticsProvider.tsx

```diff
@@ -497,32 +493,38 @@ export const AnalyticsProvider: React.FC<{ children: React.ReactNode }> = ({
   function getAllowListAnswers(
     nodeId: string,
     breadcrumb: Store.userData,
-  ): Record<string, unknown>[] | undefined {
+  ): Partial<Record<AllowListKey, any>> | undefined {
     const answers = getAnswers(nodeId);
     const data = getData(breadcrumb);
 
-    const allowListAnswers = [...answers, ...data];
-    if (!allowListAnswers.length) return;
+    if (!answers && !data) return;
+    const allowListAnswers = Object.assign({}, answers, data);
 
     return allowListAnswers;
   }
 
   /**
    * Extract allowlist answers from user answers
-   * e.g. from Checklist or Question components
+   * e.g., from Checklist or Question components
    */
-  function getAnswers(nodeId: string) {
+  function getAnswers(
+    nodeId: string,
+  ): Partial<Record<AllowListKey, any>> | undefined {
     const { data } = flow[nodeId];
-    const nodeFn: string = data?.fn || data?.val;
-    if (!nodeFn || !ALLOW_LIST.includes(nodeFn)) return [];
+    const nodeFn: string | undefined = data?.fn || data?.val;
+
+    if (!nodeFn || !ALLOW_LIST.includes(nodeFn as AllowListKey)) return;
```

Comment:
```suggestion
       const isAllowListKey = (nodeFn: string): nodeFn is AllowListKey => {
      return (ALLOW_LIST as readonly string[]).includes(nodeFn);
    };

    if (!nodeFn || !isAllowListKey(nodeFn)) return;
```

If we add a type guard here instead of just `.includes()` we'll get type narrowing and can drop the casting to `AllowListKey` on line 526.